frontend/src/containers/ChatToolbar/index.tsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Routine } from 'redux-saga-routines';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  IconDefinition,
+  faLock,
+  faHashtag,
+  faPodcast,
+  faAt,
+  faCopy,
+  faSearch,
+  faIdCard,
+  faTh,
+  faDatabase,
+  faReply,
+  faPlus,
+  faPlay
+} from '@fortawesome/free-solid-svg-icons';
+import { IAppState } from '../../common/models/store';
+import { IWorkspaceState } from '../../scenes/Workspace/reducers/reducer';
+import { IChat } from '../../common/models/workstate/chat';
+import styles from './styles.module.sass';
+import { userChannelRoutine } from '../../scenes/Workspace/routines/routines';
+
+let linkPanel = false;
+let directPanel = false;
+
+interface IProps {
+  workstate: IWorkspaceState;
+  userChannelRoutine: Routine;
+}
+
+const ChatToolbar = ({
+  workstate,
+  userChannelRoutine: channelRoutine
+}: IProps) => {
+  const { channels, directMessages, selectedChat } = workstate;
+  // Обработчики событий
+  const doThreadSelect = (chat: IChat) => {
+    // Select Thread Channel
+    channelRoutine(chat);
+  };
+  const doMentionsSelect = (chat: IChat) => {
+    // Select Mentions & reactions Channel
+    channelRoutine(chat);
+  };
+  const doDraftSelect = (chat: IChat) => {
+    // Select Drafts Channel
+    channelRoutine(chat);
+  };
+  const doChannelBrowserSelect = (chat: IChat) => {
+    // Select Channel Browser
+    channelRoutine(chat);
+  };
+  const doPeopleGrupsSelect = (chat: IChat) => {
+    // Select People & user groups
+    channelRoutine(chat);
+  };
+  const doAppsSelect = (chat: IChat) => {
+    // Select Apps Channel
+    channelRoutine(chat);
+  };
+  const doFileBrowseSelect = (chat: IChat) => {
+    // Select File Browser
+    channelRoutine(chat);
+  };
+  const doShowLessSelect = (chat: IChat) => {
+    // Select Show less Channel
+    channelRoutine(chat);
+  };
+  const doSelectChannel = (channel: IChat) => {
+    channelRoutine(channel);
+  };
+
+  const getClassNameDiv = (state: boolean) => (
+    state
+      ? styles.listBoxHidden
+      : styles.listBox
+  );
+  const getClassNameDivDefault = (state: boolean) => (
+    state
+      ? styles.listBoxHiddenDeff
+      : styles.listBoxDeff
+  );
+  const getClassNameImg = (state: boolean) => (
+    state
+      ? styles.chanelsImgRotate
+      : styles.chanelsImg
+  );
+
+  const changeListView = () => {
+    linkPanel = !linkPanel;
+    const element = document.getElementById('divChanel') as HTMLElement;
```

Comment:
it's incorrect to work with dom here. it's better to work with flags and styles

Reply:
Done